matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/MXOlmDevice.kt

```diff
@@ -421,6 +428,34 @@ internal class MXOlmDevice @Inject constructor(
         return null
     }
 
+    fun storeOutboundGroupSessionForRoom(roomId: String, sessionId: String) {
+        outboundGroupSessionStore[sessionId]?.let {
+            store.storeCurrentOutboundGroupSessionForRoom(roomId, it)
+        }
+    }
+
+    fun restoreOutboundGroupSessionForRoom(roomId: String):  MXOutboundSessionInfo? {
+        val restoredOutboundGroupSession = store.getCurrentOutboundGroupSessionForRoom(roomId)
+        if (restoredOutboundGroupSession != null) {
+            val sessionId = restoredOutboundGroupSession.outboundGroupSession.sessionIdentifier()
+            if (!outboundGroupSessionStore.containsKey(sessionId)) {
```

Comment:
I do not understand why you are checking on `outboundGroupSessionStore` (which is now actually a cache of what we have in store, no? Rename to `outboundGroupSessionCache`?).
Can you add some comment please?

Reply:
updated the code